rotatelogs.go

```diff
@@ -279,7 +279,11 @@ func (rl *RotateLogs) rotate_nolock(filename string) error {
 
 	if rl.linkName != "" {
 		tmpLinkName := filename + `_symlink`
-		if err := os.Symlink(filename, tmpLinkName); err != nil {
+		linkDest := filename
+		if filepath.Dir(rl.linkName) == filepath.Dir(filename) {
```

Comment:
`/foo/bar/baz/file.log` and `/foo/bar/baz/symlink.log` would pass this clause, but not `/foo/bar/baz/file.log` and `/foo/bar/baz/quux/symlink.log` wouldn't. Is this what you want? (as of this writing, I'm not saying that this is a bad thing, but am just checking if this is your intention)

Reply:
yes, it is my intention, since it is a particular use-case. I think just when the log and the link file in the same dir, linking to relative path.